lib/openai/client.rb

```diff
@@ -53,12 +53,34 @@ def models
       @models ||= OpenAI::Models.new(client: self)
     end
 
+    def assistants
+      @assistants ||= OpenAI::Assistants.new(client: self)
+    end
+
+    def threads
+      @threads ||= OpenAI::Threads.new(client: self)
+    end
+
+    def messages
+      @messages ||= OpenAI::Messages.new(client: self)
+    end
+
+    def runs
+      @runs ||= OpenAI::Runs.new(client: self)
+    end
+
     def moderations(parameters: {})
       json_post(path: "/moderations", parameters: parameters)
     end
 
     def azure?
       @api_type&.to_sym == :azure
     end
+
+    def beta(apis)
+      dup.tap do |client|
+        client.send(:add_headers, "OpenAI-Beta": apis.map { |k, v| "#{k}=#{v}" }.join(";"))
```

Comment:
`add_headers` is a public method.  Prefer calling it explicitly.
```suggestion
        client.add_headers "OpenAI-Beta": apis.map { |k, v| "#{k}=#{v}" }.join(";")
```

Reply:
Good point. I didn't actually intend to make the other header methods public so I've changed the others back to private.